src/Control/Effect/NonDet.hs

```diff
@@ -12,10 +14,13 @@ module Control.Effect.NonDet
 import Control.Applicative (Alternative(..))
 import Control.Effect.Choose hiding ((<|>), many, some)
 import Control.Effect.Empty hiding (empty, guard)
+import Control.Effect.Sum
 import Control.Monad (guard)
 import Data.Coerce
 import Data.Monoid (Alt(..))
 
+type NonDet = Empty :+: Choose
```

Comment:
Is there a performance hit when we change to `Empty :+: Choose`?

Reply:
Looks like no, or at least not noticeably.

```
benchmarked NonDet/N-queens problem/4
time                 10.90 μs   (10.12 μs .. 11.87 μs)
                     0.975 R²   (0.957 R² .. 0.997 R²)
mean                 10.84 μs   (10.70 μs .. 11.17 μs)
std dev              689.8 ns   (455.6 ns .. 1.066 μs)
variance introduced by outliers: 39% (moderately inflated)

benchmarked NonDet/N-queens problem/8
time                 636.3 μs   (626.4 μs .. 648.0 μs)
                     0.997 R²   (0.994 R² .. 0.999 R²)
mean                 643.8 μs   (638.0 μs .. 653.1 μs)
std dev              25.68 μs   (18.25 μs .. 42.10 μs)
variance introduced by outliers: 22% (moderately inflated)

benchmarking NonDet/N-queens problem/16 ... took 11.79 s, total 56 iterations
benchmarked NonDet/N-queens problem/16
time                 209.0 ms   (200.2 ms .. 221.0 ms)
                     0.996 R²   (0.992 R² .. 1.000 R²)
mean                 214.9 ms   (211.4 ms .. 217.0 ms)
std dev              4.892 ms   (2.475 ms .. 7.680 ms)
```

After:

```
benchmarked NonDet/N-queens problem/4
time                 10.54 μs   (10.37 μs .. 10.68 μs)
                     0.998 R²   (0.997 R² .. 0.999 R²)
mean                 10.38 μs   (10.31 μs .. 10.44 μs)
std dev              244.9 ns   (203.3 ns .. 295.2 ns)

benchmarked NonDet/N-queens problem/8
time                 644.4 μs   (630.4 μs .. 667.1 μs)
                     0.993 R²   (0.981 R² .. 0.999 R²)
mean                 641.6 μs   (635.8 μs .. 652.8 μs)
std dev              27.19 μs   (16.27 μs .. 47.32 μs)
variance introduced by outliers: 23% (moderately inflated)

benchmarking NonDet/N-queens problem/16 ... took 11.93 s, total 56 iterations
benchmarked NonDet/N-queens problem/16
time                 219.0 ms   (211.2 ms .. 223.1 ms)
                     0.999 R²   (0.996 R² .. 1.000 R²)
mean                 216.7 ms   (214.2 ms .. 218.9 ms)
std dev              4.006 ms   (2.323 ms .. 5.244 ms)
```

Reply:
Yeah, basically `inj` is not the bottleneck.